### PR TITLE
Set the Copyright and Licensing info in all the files

### LINKED ISSUE #827: Add [SPDX](https://spdx.dev/ids/) identifiers in the source code

### PR DESCRIPTION
**Pull request by abompard**
Make use of the [reuse](https://reuse.software) tool.

Also setup pre-commit to check all the files upon commit, with the following hooks:

- some basic hooks provided by pre-commit itself
- `black`
- `ruff`
- `reuse` for licenses

### CODE REVIEW

File: LICENSES/LicenseRef-Fedora-trademark.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Fedora®, the Fedora word design, and the Infinity design logo, either separately or in combination, are hereinafter referred to as "Fedora Trademarks" and are
+trademarks of Red Hat, Inc., registered in the United States and other countries.
+
+This images includes the Fedora Trademarks.  You are granted the right to use the Package during the normal operation of software spins approved by the Fedora
```
Comment (nphilipp) [inline LICENSES/LicenseRef-Fedora-trademark.txt]: Should probably be singular: “This image includes …”

File: .coveragerc [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# SPDX-FileCopyrightText: 2023 Fedora Infrastructure <admin@fedoraproject.org>
```
Comment (nphilipp) [inline .coveragerc]: I’m not sure it should be like this. As far as I know, neither Fedora nor the Infrastructure Team are legal entities who can hold a copyright.

File: DCO.txt
Comment (nphilipp) [inline DCO.txt:1]: Why is this file removed?

File: frontend/package-lock.json.license
Comment (nphilipp) [inline frontend/package-lock.json.license:1]: I wonder if adding a copyright statement to an autogenerated lock file isn’t a bit overboard…

File: tests/data/jsonschema_hyperschema.json.license
Comment (nphilipp) [inline tests/data/jsonschema_hyperschema.json.license:1]: I guess these 👆👇 shouldn’t have a copyright header at all, they’re autogenerated API schemas downloaded from services partly outside of Fedora.

**Review by nphilipp — CHANGES_REQUESTED** [review]

File: frontend/package-lock.json.license
Comment (abompard) [reply frontend/package-lock.json.license:1]: It is, but the tool insists that all files should have a license, and who am I to argue with lawyers? :-)

That said, I'll see if I can get away with just the license and no copyright.

File: DCO.txt
Comment (abompard) [reply DCO.txt:1]: I've replaced it with a link to developercertificate.org. Or at least I thought I did, but it was overwritten my one of my many `git restore`... Fixing.

File: frontend/package-lock.json.license
Comment (abompard) [reply frontend/package-lock.json.license:1]: Yep, I just noted that it's autogenerated.

File: tests/data/jsonschema_hyperschema.json.license
Comment (abompard) [reply tests/data/jsonschema_hyperschema.json.license:1]: I've replaced the copyright by a statement that it's autogenerated

File: .coveragerc (hunk 2 above)
Comment (abompard) [reply .coveragerc]: Fixed!

File: LICENSES/LicenseRef-Fedora-trademark.txt (hunk 1 above)
Comment (abompard) [reply LICENSES/LicenseRef-Fedora-trademark.txt]: Fixed, thanks.

File: LICENSES/MIT.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+MIT License
+
+Copyright (c) <year> <copyright holders>
```
Comment (nphilipp) [inline LICENSES/MIT.txt:3]: I think this should be `Copyright Contributors to the Fedora Project`.

File: tox.ini [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
+; SPDX-License-Identifier: MIT
+
 [tox]
 minversion = 3.10.0
 envlist = py310,py311,docs,black,lint
 isolated_build = true
 skip_missing_interpreters = true
 
 [testenv]
+passenv = HOME
```
Comment (nphilipp) [inline tox.ini:12]: What’s this for? Running `tox` works for me without it.

**Review by nphilipp — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 4 above)
Comment (nphilipp) [reply tox.ini:12]: Per convo in the chat: this makes running ruff as a pre-commit linter work.

File: LICENSES/MIT.txt (hunk 3 above)
Comment (abompard) [reply LICENSES/MIT.txt:3]: Actually, [they recommend against doing that](https://reuse.software/faq/#license-templates). Weird, I know.

Comment (nphilipp) [reply LICENSES/MIT.txt:3]: Okay then 😂

**Review by nphilipp — APPROVED** [review]
Looks good to me!

**Pull request merged**